src/pandas_profiling/__init__.py

```diff
@@ -4,48 +4,107 @@
 """
 import json
 import warnings
-from pathlib import Path
 from datetime import datetime
+from pathlib import Path
+from typing import Union
 
-import pandas as pd
+import joblib
```

Comment:
It seems this dependency is only used to hash the DataFrame. Is this really necessary?

These might be relevant:
http://drorata.github.io/posts/2017/May/26/when-trying-to-hash-a-data-frame/index.html
https://github.com/pandas-dev/pandas/issues/16517#issuecomment-304364225